articles/architecture-scenarios/implementations/mobile-api/mobile-implementation-android.md

```diff
@@ -0,0 +1,1260 @@
+            e.printStackTrace();
+        }
+        String postStr = postBody.toString();
+
+        final Request.Builder reqBuilder = new Request.Builder()
+                .post(RequestBody.create(MEDIA_TYPE_JSON, postStr))
+                .url(getString(R.string.api_url))
+                .addHeader("Authorization", "Bearer " + accessToken);
+
+        OkHttpClient client = new OkHttpClient();
+        Request request = reqBuilder.build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Request request, IOException e) {
+                Log.e("API", "Error: ", e);
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        Toast.makeText(FormActivity.this, "An error occurred", Toast.LENGTH_SHORT).show();
+                    }
+                });
+            }
+
+            @Override
+            public void onResponse(final Response response) throws IOException {
+                final String resBody = response.body().string();
+
+                runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (response.isSuccessful()) {
+                            clearForm((ViewGroup) findViewById(R.id.mainForm));
+                            Intent intent = new Intent(FormActivity.this, TimeSheetActivity.class);
+                            FormActivity.this.startActivity(intent);
+                        } else {
+                            Toast.makeText(FormActivity.this, "Timesheet creation failed.", Toast.LENGTH_SHORT).show();
+                        }
+                    }
+                });
+            }
+        });
+    }
+
+    private void clearForm(ViewGroup group) {
+        for (int i = 0, count = group.getChildCount(); i < count; ++i) {
+            View view = group.getChildAt(i);
+            if (view instanceof EditText) {
+                ((EditText)view).setText("");
+            }
+
+            if(view instanceof ViewGroup && (((ViewGroup)view).getChildCount() > 0))
+                clearForm((ViewGroup)view);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.form_action_menu, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.action_view:
+                startActivity(new Intent(FormActivity.this, TimeSheetActivity.class));
+                break;
+            case R.id.action_profile:
+                startActivity(new Intent(FormActivity.this, UserActivity.class));
+                break;
+            default:
+                return super.onOptionsItemSelected(item);
+
+        }
+        return true;
+    }
+}
+```
```

Comment:
I didn't read this in detail yet but make sure we tell them how to test it, and that you add a closing sentence at the very end of the doc, sth like `That's it! You are done!`